File: go.ingestor/redis_store.go
package main

import (
	"context"
	"encoding/json"
	"fmt"

	"github.com/redis/go-redis/v9"
)

var ctx = context.Background()

func StoreCandle(rdb *redis.Client, candle Candle) error {
	key := fmt.Sprintf("candle:%s:%s:%d", candle.Ticker, candle.Interval, candle.Timestamp)

	data, err := json.Marshal(candle)
	if err != nil {
		return err
	}

	return rdb.Set(ctx, key, data, 0).Err()
}

File: ingestor/src/infra/redis_store.py
import json
import redis


async def store_candles(rdb: redis.Redis, candles):
    pipe = rdb.pipeline()
    for c in candles:
        key = f"candle:{c.ticker}:{c.timeframe}:{c.timestamp}"
        pipe.set(key, json.dumps(c.to_dict()))
        pipe.expire(key, 60*60)
    await pipe.execute()